src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -197,34 +197,43 @@ public void Start()
                 }
 
                 // STEP 2: Open the socket connection to the remote host
-                var cts = new CancellationTokenSource();
-                var tok = cts.Token;
-                
-                tok.Register(() =>
-                {
-                    _client.Close();
-                });
+                var cts = new CancellationTokenSource(ConnectTimeout);
+                cts.CancelAfter(ConnectTimeout);
 
+                using (cts.Token.Register(() => { _client.Close(); }))
+                {
+                    _client.Dispose();
+                }
                 try
                 {
-                    _client.ConnectAsync(_logic.UrlRequest.Host, _logic.UrlRequest.Port).ContinueWith(t =>
+                    using (cts)
                     {
```

Comment:
I don't think `using` can be used here because the token source needs to remain valid asynchronously during the connect timeout period.